enclave/src/rpc/api.rs

```diff
@@ -71,17 +108,47 @@ where
 
     fn validate_transaction(
         &self,
-        _at: &BlockId<Self::Block>,
         _source: TrustedOperationSource,
         uxt: StfTrustedOperation,
+        shard: ShardIdentifier,
     ) -> Self::ValidationFuture {
         let operation = match uxt {
-            StfTrustedOperation::direct_call(call) => ValidTransaction {
-                priority: 1 << 20,
-                requires: vec![],
-                provides: vec![vec![call.nonce as u8], call.signature.encode()],
-                longevity: 3,
-                propagate: true,
+            StfTrustedOperation::direct_call(signed_call) => {
+                let nonce = signed_call.nonce;
+                let from = signed_call.call.account();
+
+                let expected_nonce = match expected_nonce(shard, &from) {
+                    Ok(nonce) => nonce,
+                    Err(_) => return Box::pin(ready(Ok(Err(TransactionValidityError::Unknown(
+                        UnknownTransaction::CannotLookup,
+                    ))))),
+                };
+                if nonce < expected_nonce {
+                    return Box::pin(ready(Ok(Err(TransactionValidityError::Invalid(
+                        InvalidTransaction::Stale
+                    )))))
+                }
+                if nonce > expected_nonce + 64 {
```

Comment:
why a hardcoded value?

Reply:
moved to constants - still hard coded though. Or what were you thinking of?

Reply:
what is the reasoning for 64?

Reply:
This conversation is obsolete with the trustet getter for getting the nonce